src/main/java/it/gov/pagopa/logextractor/util/external/filestorage/CustomFileOutputStream.java

```diff
@@ -0,0 +1,24 @@
+package it.gov.pagopa.logextractor.util.external.filestorage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+public class CustomFileOutputStream extends FileOutputStream{
+
+	private String fileName;
+	
+	public CustomFileOutputStream(String filename) throws FileNotFoundException {
+		super(filename+".tmp");
+		this.fileName = filename;
+	}
+
+	@Override
+	public void close() throws IOException {
+		super.close();
+		new File(fileName+".tmp").renameTo(new File(fileName));
```

Comment:
qui abbiamo Sonar ci riporta un bug: https://sonarcloud.io/project/issues?resolved=false&types=BUG&pullRequest=79&id=pagopa_pn-logextractor-be&open=AYlUskGrR6RvgOjah60A.

Se serve solo rinominare il file, possiamo fare una cosa del genere? (da testare con un jUnit):

```
Path fileTmp = Path.of(fileName);
Files.move(fileTmp, fileTmp.resolveSibling(fileName));
```

(https://mkyong.com/java/how-to-rename-file-in-java/)

Reply:
Si tratta di classi per test locali (per non usare S3), il presunto bug è che il codice ignora l'esito del rename.
Ora gestito per tener buono SonarLint e loggata eccezione.